pkg/virt-handler/device-manager/socket_device.go

```diff
@@ -212,6 +212,14 @@ func (dpi *SocketDevicePlugin) Allocate(ctx context.Context, r *pluginapi.Alloca
 	response := pluginapi.AllocateResponse{}
 	containerResponse := new(pluginapi.ContainerAllocateResponse)
 
+	prSockDir, err := safepath.JoinAndResolveWithRelativeRoot("/", dpi.socketDir)
+	if err != nil {
+		return nil, fmt.Errorf("error opening the directory %s: %v", dpi.socketDir, err)
+	}
+	err = safepath.ChownAtNoFollow(prSockDir, util.NonRootUID, util.NonRootUID)
+	if err != nil {
+		return nil, fmt.Errorf("error setting directory ownership %s: %v", dpi.socketDir, err)
+	}
```

Comment:
I have a feeling that we may not need to chown the directory at all when running pr-helper as root. It should have `0755` by default, thus it's world-readable. Will do a quick test...

Reply:
Yep, no error on unlink if the directory is not chown'ed:
```
$ k exec -ti -n kubevirt virt-handler-wxf5m -c pr-helper -- bash
bash-5.1# ls -la /var/run/kubevirt/daemons/pr/
total 0
drwxr-xr-x. 2 root root 60 Aug  7 08:44 .
drwxr-xr-x. 3 root root 16 Aug  7 08:44 ..
srwxr-xr-x. 1 root root  0 Aug  7 08:44 pr-helper.sock
bash-5.1# 
bash-5.1# /usr/bin/qemu-pr-helper -k /var/run/kubevirt/daemons/pr/pr-helper2.sock
^Cbash-5.1# 
bash-5.1# 
bash-5.1# ls -la /var/run/kubevirt/daemons/pr/
total 0
drwxr-xr-x. 2 root root 60 Aug  7 08:44 .
drwxr-xr-x. 3 root root 16 Aug  7 08:44 ..
srwxr-xr-x. 1 root root  0 Aug  7 08:44 pr-helper.sock
```

Reply:
```
Ran 3 of 1520 Specs in 757.019 seconds
SUCCESS! -- 3 Passed | 0 Failed | 2 Pending | 1515 Skipped
```